pymc_experimental/model/marginal_model.py

```diff
@@ -870,3 +932,68 @@ def step_alpha(logp_emission, log_alpha, log_P):
     # return is the joint probability of everything together, but PyMC still expects one logp for each one.
     dummy_logps = (pt.constant(0),) * (len(values) - 1)
     return joint_logp, *dummy_logps
+
+
+@_logprob.register(QMCMarginalNormalRV)
+def qmc_marginal_rv_logp(op, values, *inputs, **kwargs):
+    # Clone the inner RV graph of the Marginalized RV
+    marginalized_rvs_node = op.make_node(*inputs)
+    # The MarginalizedRV contains the following outputs:
+    # 1. The variable we marginalized
+    # 2. The dependent variables
+    # 3. The updates for the marginalized and dependent variables
+    marginalized_rv, *inner_rvs_and_updates = clone_replace(
+        op.inner_outputs,
+        replace={u: v for u, v in zip(op.inner_inputs, marginalized_rvs_node.inputs)},
+    )
+    inner_rvs = inner_rvs_and_updates[: (len(inner_rvs_and_updates) - 1) // 2]
+
+    marginalized_vv = marginalized_rv.clone()
+    marginalized_rv_node = marginalized_rv.owner
+    marginalized_rv_op = marginalized_rv_node.op
+
+    # GET QMC draws from the marginalized RV
+    # TODO: Make this an Op
+    rng = marginalized_rv_op.rng_param(marginalized_rv_node)
+    shape = constant_fold(tuple(marginalized_rv.shape))
+    size = np.prod(shape).astype(int)
+    n_draws = 2**op.qmc_order
+
+    # TODO: Wrap Sobol in an Op so we can control the RNG and change whenever
+    qmc_engine = scipy.stats.qmc.Sobol(d=size, seed=rng.get_value(borrow=False))
+    uniform_draws = qmc_engine.random(n_draws).reshape((n_draws, *shape))
+
+    if isinstance(marginalized_rv_op, MvNormal):
```

Comment:
how about using multipledispatch on this? Could be in a different PR. The rationale is that one can implement a new distribution without hacking pmx

Reply:
Most cases default to inverse icdf implementation, just mvnormal needs special case because it has no icdf and yet is probably pretty common

Reply:
I rather not make everything dispatch until it's shown to actually be necessary. Feels like premature modularity if you will